src/main/kotlin/app/revanced/patches/youtube/misc/video/information/patch/VideoInformationPatch.kt

```diff
@@ -185,5 +214,30 @@ class VideoInformationPatch : BytecodePatch(
             TIME(2),
             HIGH_PRECISION_TIME(0),
         }
+
+        private fun getReference(instructions: List<BuilderInstruction>, offset: Int, opcode: Opcode) =
+            instructions[instructions.indexOfFirst { it.opcode == opcode } + offset].reference
+
+        val Instruction.reference get() = (this as ReferenceInstruction).reference.toString()
+
+        private lateinit var speedSelectionInsertMethod: MutableMethod
+        private var speedSelectionInsertIndex = 0
+        private var speedSelectionValueRegister = 0
+
+        /**
+         * Hook the video speed selected by the user.
+         */
+        internal fun userSelectedVideoSpeedHook(targetMethodClass: String, targetMethodName: String) =
+            speedSelectionInsertMethod.addInstruction(
+            speedSelectionInsertIndex++,
+            "invoke-static {v$speedSelectionValueRegister}, $targetMethodClass->$targetMethodName(F)V"
+        )
+        
+        /**
+         * Used by [RememberPlaybackSpeedPatch]
+         */
+        internal lateinit var setPlaybackSpeedContainerClassFieldReference: String
+        internal lateinit var setPlaybackSpeedClassFieldReference: String
+        internal lateinit var setPlaybackSpeedMethodReference: String
```

Comment:
[Kotlin suggests a specific class layout](https://kotlinlang.org/docs/coding-conventions.html#class-layout), but here, it is neglected. This could be intentional, due to the lack of [this](https://github.com/revanced/revanced-patches/pull/1843/files#r1156440839), which is why fields and methods are grouped together. If it is not intentional, the coding conventions should be followed.